Podverse/ClipsListContainerViewController.swift

```diff
@@ -142,30 +166,60 @@ extension ClipsListContainerViewController:UITableViewDelegate, UITableViewDataS
     }
     
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        return 100
+        return UITableViewAutomaticDimension
+    }
+    
+    func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableViewAutomaticDimension
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-
-        let cell = tableView.dequeueReusableCell(withIdentifier: "mediaPlayerClipCell", for:indexPath) as! MediaPlayerClipTableViewCell
         
-        cell.contentView.layoutMargins.top = 16
-        cell.contentView.layoutMargins.bottom = 16
+        //if filterTypeSelected == .episode {
+            
+        //} else if filterTypeSelected == .podcast {
+            
+        //} else {
+            
+        //}
         
         let clip = clipsArray[indexPath.row]
         
-        if let title = clip.title {
-            cell.clipTitle.text = title
-        }
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath) as! ClipTableViewCell
+        
+        cell.podcastTitle?.text = clip.podcastTitle
+        cell.episodeTitle?.text = clip.episodeTitle
+        cell.clipTitle?.text = clip.title
+        
+        var time: String?
         
         if let startTime = clip.startTime {
-            cell.startTime.text = startTime.toMediaPlayerString()
+            if let endTime = clip.endTime {
+                if endTime > 0 {
+                    time = startTime.toMediaPlayerString() + " to " + endTime.toMediaPlayerString()
+                }
+            } else {
+                time = "Starts:" + startTime.toMediaPlayerString()
+            }
         }
         
-        if let endTime = clip.endTime {
-            cell.endTime.text = endTime.toMediaPlayerString()
+        if let time = time {
+            cell.time?.text = time
         }
         
+        if let episodePubDate = clip.episodePubDate {
+            cell.episodePubDate?.text = episodePubDate.toShortFormatString()
+        }
+        
+        Podcast.retrievePodcastImage(podcastImageURLString: clip.podcastImageUrl) { (podcastImage) -> Void in
+            DispatchQueue.main.async {
```

Comment:
I believe the completion is already being returned on the main queue and if it isn't, it should :)